frappe/templates/includes/comments/comments.py

```diff
@@ -18,19 +18,23 @@
 @frappe.whitelist(allow_guest=True)
 @rate_limit(key="reference_name", limit=get_comment_limit, seconds=60 * 60)
 def add_comment(comment, comment_email, comment_by, reference_doctype, reference_name, route):
-	doc = frappe.get_doc(reference_doctype, reference_name)
+	if frappe.session.user == "Guest":
+		if reference_doctype not in ("Blog Post", "Web Page"):
+			return
+
+		if not frappe.db.get_single_value("Blog Settings", "allow_guest_to_comment"):
```

Comment:
What if the comment is on a Web Page though? Not sure if this setting applies there 🤔

Reply:
Perhaps we need another setting or to refactor this setting to apply to web pages as well. Currently, there's some code to indicate that guests are always allowed for web page comments:

https://github.com/frappe/frappe/blob/40ad98359896bfdbe263fc7238334ad7e11e023f/frappe/website/doctype/web_page/web_page.py#L57